android/modules/ui/src/java/ti/modules/titanium/ui/widget/TiUIButton.java

```diff
@@ -84,17 +83,8 @@ public void processProperties(KrollDict d)
 		}
 		if (d.containsKey(TiC.PROPERTY_TITLE)) {
 			btn.setText(d.getString(TiC.PROPERTY_TITLE));
-			ViewTreeObserver vto = btn.getViewTreeObserver();
-			vto.addOnGlobalLayoutListener(new ViewTreeObserver.OnGlobalLayoutListener() {
-
-				@Override
-				public void onGlobalLayout() {
-					if(btn.getLineCount()>1 && btn.getMeasuredHeight() > btn.getLineHeight()){
-						btn.setEllipsize(TextUtils.TruncateAt.END);
-						btn.setMaxLines(1);
-					}
-				}
-			});
+			btn.setEllipsize(TextUtils.TruncateAt.MIDDLE);
+			btn.setMaxLines(1);
```

Comment:
Can you move these 2 lines into the TiUIButton constructor please?
(Before the `setNativeView(btn)` call.)

This way these setting will always be applied. This is in case the "title" property is not set upon button creation and is set afterwards instead.

Other than that, yeah, it should be this simple. :)